app/controllers/ileQuery/IleQueryController.scala

```diff
@@ -0,0 +1,81 @@
+package controllers.ileQuery
+
+import connectors.CustomsDeclareExportsMovementsConnector
+import connectors.exchanges.IleQueryExchange
+import controllers.actions.AuthenticatedAction
+import controllers.exchanges.AuthenticatedRequest
+import forms.IleQuery
+import javax.inject.{Inject, Singleton}
+import models.UcrBlock
+import models.cache.Answers
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, Cookie, DiscardingCookie, MessagesControllerComponents, Result}
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import utils.FieldValidator.validDucr
+import views.html.ile_query
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class IleQueryController @Inject()(
+  authenticate: AuthenticatedAction,
+  mcc: MessagesControllerComponents,
+  connector: CustomsDeclareExportsMovementsConnector,
+  ileQueryPage: ile_query
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
+
+  def displayPage(): Action[AnyContent] = authenticate { implicit request =>
+    Ok(ileQueryPage(IleQuery.form))
+  }
+
+  def submit(): Action[AnyContent] = authenticate { implicit request =>
+    IleQuery.form.fold(
+      formWithErrors => BadRequest(ileQueryPage(formWithErrors)),
+      validUcr => Redirect(controllers.ileQuery.routes.IleQueryController.submitQuery(validUcr))
+    )
+  }
+
+  def submitQuery(ucr: String): Action[AnyContent] = authenticate.async { implicit request =>
+    request.cookies.get("conversationId") match {
+      case Some(cookie) =>
+        connector.fetchQueryNotification(cookie.value, request.providerId).map { notifications =>
+          if(notifications.isEmpty) Ok("loading").withHeaders("refresh" -> "1")
```

Comment:
It would be good to be able to controll the refresh rate. At least through the config.